### ISSUE TITLE
Alerting: SMTP emails are not working with AWS SES

### ISSUE THREAD

**Issue by j0rzsh** [grafana/grafana#72096]
### What happened?

We have the setup working previously and suddenly it stopped to work.
We are using AWS SES for sending emails.

We notice this when we were testing the upgrade to 10.0.1, but they were failing previously and we won't notice.
We have treid 9.0.5, 9.5.6, 10.0.1 and 10.0.2. The same error appears:

```
2023-07-21T12:27:49.630+02:00 | logger=ngalert.notifier.email t=2023-07-21T10:27:49.62998238Z level=error msg="Missing receiver"
2023-07-21T12:27:49.630+02:00 | logger=ngalert.notifier.email t=2023-07-21T10:27:49.63000889Z level=error msg="Missing group labels"
```

Doing the same test locally:

```
ERROR[07-21|10:22:14] Missing receiver                         logger=ngalert.notifier.email notifierUID=
ERROR[07-21|10:22:14] Missing group labels                     logger=ngalert.notifier.email notifierUID=
```

Setting log level to debug does not show anything else related to that.

In the UI, the popup appears as correct email sent.

We have this settings related to SMTP:

```
GF_SMTP_ENABLED=true
GF_SMTP_FROM_ADDRESS=email@domain.com
GF_SMTP_FROM_NAME=Grafana
GF_SMTP_HOST=email-smtp.eu-west-1.amazonaws.com:587
GF_SMTP_SKIP_VERIFY=false
GF_SMTP_STARTTLS_POLICY=MandatoryStartTLS
GF_SMTP_USER="AAAAAA"
GF_SMTP_PASSWORD="BBBBBBB"
```

### What did you expect to happen?

The email should be sent.

### Did this work before?

Yes. Was working on the same version that now is not (9.0.5 in our case).

### How do we reproduce it?

1. Enable SMTP in grafana using AWS SES
2. Create a contact point
3. Try to send a test email

### Environment (with versions)?

Grafana:
OS:
Browser:

### Grafana platform?

None

**Comment by usmangt** [comment]
Hi @j0rzsh,

Thanks for opening this issue.

Try to use `startTLS_policy = NoStartTLS` as found a [solution](https://community.grafana.com/t/level-error-msg-missing-receiver/74114) on the community forum.

Let us know if this helps or not.

**Comment by j0rzsh** [comment]
@usmangt Hi! I've already tried that I forgot to mention it. It does not work at least with AWS SES. It throws an error saying that you cannot send the mail using an unciphered connection. Thanks anyway!

**Comment by armandgrillet** [comment]
```
2023-07-21T12:27:49.630+02:00 | logger=ngalert.notifier.email t=2023-07-21T10:27:49.62998238Z level=error msg="Missing receiver"
2023-07-21T12:27:49.630+02:00 | logger=ngalert.notifier.email t=2023-07-21T10:27:49.63000889Z level=error msg="Missing group labels"
```

These two log lines do not affect the testing of the contact point, they are coming from the upstream Alertmanager and an email could still be sent even if these logs are displayed. 

However, you seem to not receive emails anymore. Is there another error message you get when the contact point is failing? There should be a dedicated log line with a description of what went wrong there.

**Comment by j0rzsh** [comment]
Hi @armandgrillet. Nothing else is showing up in the logs.

I've set GF_LOG_LEVEL to debug and this was being printed:

```
DEBUG[08-01|07:55:32] using cached permissions                 logger=accesscontrol.service key=rbac-permissions-1-user-1
ERROR[08-01|07:55:32] Missing receiver                         logger=ngalert.notifier.email notifierUID=
ERROR[08-01|07:55:32] Missing group labels                     logger=ngalert.notifier.email notifierUID=
DEBUG[08-01|07:55:36] Recording state cache metrics            logger=ngalert.state.manager now=2023-08-01T07:55:36.937920709Z
```

This is happening both in our AWS environment and in my local environment. I have also added my personal email instead of my company one into AWS SES to check if the problem is our Google Workspace config or something like that. With this configuration it is also failing.

Whenever I have my personal email unverified in AWS SES, this is appearing in the web ui (nothing in the logs):

```
Failed to send test alert.: failed to send notification to email addresses: myemail@gmail.com: gomail: could not send email 1: 554 Message rejected: Email address is not verified. The following identities failed the check in region AWS-REGION: myemail@gmail.com
```

**Comment by j0rzsh** [comment]
Hi again @armandgrillet @usmangt 
We have discovered the root cause of this, it is not related to Grafana.

I will explain in case anyone has the same problem:

Our issue was caused by our DMARC configuration. We are using an smtp user in AWS in our root account. In that root account we have our domain verified in AWS SES (DKIM config).

For grafana and other tools, we have a dedicated account and we had a dedicated user in that account. In that account we don't have the domain verified. When we have changed the DMARC configuration to be strict, set to reject and with proper smtp domains added into our SPF policy. Before making that change, it was properly working.

I have used the credentials from the first user in Grafana and it is working, so the problem is in our configuration.

Many thanks both of you for the help!
